dynamiqs/hermitian_conjugate.py

```diff
@@ -0,0 +1,23 @@
+from jaxtyping import ArrayLike
+from qutip import Qobj
+
+from .qarrays import QArray, QArrayLike, asqarray
+
+
+class HermitianConjugate:
+    def __radd__(self, y: QArrayLike) -> QArrayLike:
+        if isinstance(y, Qobj):
+            y = asqarray(y)
+
+        if isinstance(y, QArray):
+            return y + y.dag()
+        elif isinstance(y, ArrayLike):
```

Comment:
I think the CI complains because of this. I'm not sure but we're not supposed to check types using `jaxtyping` (although that's also what we do at other places in the library). @pierreguilmin an idea?

Reply:
See also the equinox functions: https://docs.kidger.site/equinox/api/manipulation/#equinox.is_array

Reply:
Just need to do: (for Python 3.9 I think)
```
from typing import get_args
isinstance(y, get_args(ArrayLike))
```

Reply:
Thanks for the tip!